### ISSUE TITLE
Sqlite3 dependency

### ISSUE THREAD

**Issue by Leon0402** [fnc12/sqlite_orm#571]
Hi there, 

you currently require people to link against sqlite3, this is something the lib itself should rather to. 

Sqlite3 itsel doesn't provide any cmake integration, but cmake does https://cmake.org/cmake/help/v3.14/module/FindSQLite3.html. 

Currently I do the following to use your lib. I pull it in via FetchContent and then I do

```
find_package(SQLite3 REQUIRED)
target_link_libraries(MyTarget Private sqlite_orm SQLite::SQLite3)
```

It should rather be just 
```
target_link_libraries(MyTarget Private sqlite_orm)
```

To achieve that:
Put a `find_package(SQLite3 REQUIRED)` somewhere before you add your targets, but it needs to be in the same file (scope) or in the parent scope, as targets exposed by find_package are not global (compared to FetchContent).

Then link against sqlite
```
target_link_libraries(sqlite_orm PUBLIC SQLite::SQLite3)
```

**Important**: It should most likely be public as the consumer (me) needs to link against sqlite3 as well ... so this will basically add sqlite3 to my targets, which link against your target.

Consider also to provide an alias target for sqlite_orm, so one can link against sqlite_orm::sqlite_orm .. or some name you like. This has a few advantages
- Less likely to give a name clash (like if another lib has a target sqlite_orm) I use
- Linking of the lib is consistent, no matter if you use find_package, add_subdirectory or fetchContent
- Cmake know that it is a target! If I write `target_link_libraries(MyTarget Private sqlite_orm)` cmake doesn't know if sqlite_orm is a target or a lib ... and therefore the error message if not found, will be worse.

Also consider saying that you do support FetchContent. I just tried it and it works quite well. It will work even better if you make sqlite3 a dependency in cmake and catch2 as well (see other post)

**Comment by fnc12** [comment]
Hi. It is very nice that there is someone who knows cmake well enough. May I ask you to make a branch from `dev`, commit these improvements and make a PR to `dev`? After that you will be automatically included into contributors list on main page. I would appreciate this cause I don't know cmake as much as you do. Thank you very much

**Comment by Leon0402** [comment]
Fixed with #572 #575
